tools/cldr-code/src/main/java/org/unicode/cldr/util/CLDRFileOverride.java

```diff
@@ -0,0 +1,162 @@
+package org.unicode.cldr.util;
+
+import com.ibm.icu.util.VersionInfo;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Set;
+import org.unicode.cldr.util.XPathParts.Comments;
+
+public class CLDRFileOverride extends CLDRFile {
+
+    /**
+     * Overrides the values of the sourceFile. The keys of the overrides must already be in the
+     * XMLSource source
+     */
+    public CLDRFileOverride(CLDRFile sourceFile, Map<String, String> valueOverrides) {
+        super(new DelegateXMLSource(sourceFile.dataSource, valueOverrides));
+    }
+
+    /**
+     * Overrides the values of the source. The keys of the overrides must already be in the
+     * XMLSource source
+     */
+    public static class DelegateXMLSource extends XMLSource {
```

Comment:
There's already a top-level class named DelegateXMLSource. I'm confused how this new class is related to the old one. Does it need to have the same name, and does it need to be public?

Reply:
That is in the apps directory, and inaccessible (rightfully) in the util directory. Plus, this has some changes. Will retitle.

Reply:
The class in cldr-apps could certainly be moved to util if it was useful.